### DISCUSSION TITLE
Question: How long does the integration check for the API days use

### DISCUSSION THREAD

**Discussion by JRascagneres** [BottlecapDave/HomeAssistant-OctopusEnergy#334] [Q&A]
Weird question but say Octopus API usage doesn't update for a few hours or day... How long does the integration 'check' for updates? Question is basically... How long does Octopus have to not update properly for the integration not to grab my data...

**Comment by BottlecapDave** [comment]
Hi there.

The integration downloads current usage data (if Octopus Home Mini is available) every minute, back filling some data if it failed to retrieve it previously. The integration downloads previous consumption data every 30 minutes (or every minute if data is stale). Another user previously mentioned that they experienced timeouts from OE which exceeded the default timeouts for the integration which meant their data never updated. Do you have any errors in your logs?

**Reply by JRascagneres** [reply]
Thanks. It was literally on the Octopus side just hadn't obtained my previous days usage until like midday so I just wanted to ensure it'd eventually pick it up (which it did). I was just wondering how bad would it have to get before the integration wouldn't pick it up.
